### ISSUE TITLE
Unable to generate from SQLite database

### ISSUE THREAD

**Issue by pskaarup** [etiennestuder/gradle-jooq-plugin#79]
I'm trying to setup a simple jooq, spring boot example with a sqlite database using java 10 features.

For this im using a database from [sqlitetutorial](http://www.sqlitetutorial.net/sqlite-sample-database/) specifically [chinook.zip](http://www.sqlitetutorial.net/wp-content/uploads/2018/03/chinook.zip) using below gradle configuration

```
gradle --version

------------------------------------------------------------
Gradle 4.8.1
------------------------------------------------------------

Build time:   2018-06-21 07:53:06 UTC
Revision:     0abdea078047b12df42e7750ccba34d69b516a22

Groovy:       2.4.12
Ant:          Apache Ant(TM) version 1.9.11 compiled on March 23 2018
JVM:          10.0.1 ("Oracle Corporation" 10.0.1+10)
OS:           Mac OS X 10.13.5 x86_64
```

```gradle

buildscript {
    repositories {
        mavenLocal()
        mavenCentral()
        maven {
            url "https://plugins.gradle.org/m2/"
        }
    }

    dependencies {
        // general requirements
        classpath "org.springframework.boot:spring-boot-gradle-plugin:2.0.3.RELEASE"
        classpath group: 'org.xerial', name: 'sqlite-jdbc', version: '3.21.0'

        // gradle-jooq-plugin requirement
        classpath 'nu.studer:gradle-jooq-plugin:3.0.0' // 3.0.1 can't find JAXB even though it's there -> https://github.com/etiennestuder/gradle-jooq-plugin/issues/77
        classpath 'com.sun.xml.bind:jaxb-core:2.3.0.1'
        classpath 'com.sun.xml.bind:jaxb-impl:2.3.0.1'
        classpath 'javax.activation:activation:1.1.1'
        classpath 'javax.xml.bind:jaxb-api:2.3.0'

    }
}

plugins {
    id 'java'
    id 'nu.studer.jooq' version '3.0.0'
}

apply plugin: 'java'
apply plugin: 'idea'
apply plugin: 'org.springframework.boot'
apply plugin: 'io.spring.dependency-management'
apply plugin: 'nu.studer.jooq'

bootJar {
    baseName = 'gs-spring-boot'
    version = '0.1.0'
}

group 'test'
version '1.0-SNAPSHOT'

repositories {
    mavenCentral()
    jcenter()
}


sourceCompatibility = 10

dependencies {
    // =============== dev dependencies ========================
    // Spring
    compile 'org.springframework.boot:spring-boot-starter-web'

    // Database
    compile 'org.jooq:jooq:3.11.3'
    compile group: 'org.xerial', name: 'sqlite-jdbc', version: '3.21.0'

    // Jooq Spring
    compile group: 'org.springframework.boot', name: 'spring-boot-starter-jooq', version: '2.0.3.RELEASE'

    // Jooq generator requirement
    jooqRuntime group: 'org.xerial', name: 'sqlite-jdbc', version: '3.21.0'
    jooqRuntime 'org.jooq:jooq-meta:3.11.3'
    jooqRuntime 'org.jooq:jooq-codegen:3.11.3'
    // java 9+ requirement
    jooqRuntime 'javax.activation:activation:1.1.1'
    jooqRuntime 'javax.xml.bind:jaxb-api:2.3.0'
    jooqRuntime 'com.sun.xml.bind:jaxb-core:2.3.0.1'
    jooqRuntime 'com.sun.xml.bind:jaxb-impl:2.3.0.1'


    // =============== Test dependencies =======================
    testCompile group: 'junit', name: 'junit', version: '4.12'
    testCompile 'org.springframework.boot:spring-boot-starter-test'
}

jooq {
    SDE(sourceSets.main) {
        jdbc {
            driver = 'org.sqlite.JDBC'
//            http://www.sqlitetutorial.net/sqlite-sample-database/
//            http://www.sqlitetutorial.net/wp-content/uploads/2018/03/chinook.zip
            url = 'jdbc:sqlite:/<PATH TO DB>/chinook.db'
        }
        generator {
            name = 'org.jooq.codegen.DefaultGenerator'
            strategy {
                name = 'org.jooq.codegen.DefaultGeneratorStrategy'
            }
            database {
                name = 'org.jooq.meta.sqlite.SQLiteDatabase'
            }
            generate {
                relations = true
                deprecated = false
                records = true
                immutablePojos = true
                fluentSetters = true
                // Annotations used normally are not valid java 10
                generatedAnnotation = false
            }
            target {
                packageName = 'com.jooq.generated'
            }
        }
    }
}

```

# Error log
```log
./gradlew generateSDEJooqSchemaSource            

> Task :generateSDEJooqSchemaSource
Jul 15, 2018 4:08:14 PM org.jooq.tools.JooqLogger info
INFO: Initialising properties  : /Users/pvs/Documents/evedb/test/build/tmp/jooq/config.xml
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.sun.xml.bind.v2.runtime.reflect.opt.Injector (file:/Users/pvs/.gradle/caches/modules-2/files-2.1/com.sun.xml.bind/jaxb-impl/2.3.0.1/2e979dabb3e5e74a0686115075956391a14dece8/jaxb-impl-2.3.0.1.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int)
WARNING: Please consider reporting this to the maintainers of com.sun.xml.bind.v2.runtime.reflect.opt.Injector
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO: No <inputCatalog/> was provided. Generating ALL available catalogs instead.
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO: No <inputSchema/> was provided. Generating ALL available schemata instead.
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO: License parameters       
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO: ----------------------------------------------------------
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   Thank you for using jOOQ and jOOQ's code generator
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:                          
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO: Database parameters      
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO: ----------------------------------------------------------
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   dialect                : SQLITE
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   URL                    : jdbc:sqlite:/<PATH TO DB>/chinook.db
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   target dir             : /<Some path>/test/build/generated-src/jooq/SDE
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   target package         : com.jooq.generated
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   includes               : [.*]
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   excludes               : []
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   includeExcludeColumns  : false
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO: ----------------------------------------------------------
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:                          
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO: JavaGenerator parameters 
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO: ----------------------------------------------------------
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   annotations (generated): false
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   annotations (JPA: any) : false
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   annotations (JPA: version): 
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   annotations (validation): false
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   comments               : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   comments on attributes : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   comments on catalogs   : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   comments on columns    : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   comments on keys       : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   comments on links      : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   comments on packages   : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   comments on parameters : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   comments on queues     : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   comments on routines   : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   comments on schemas    : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   comments on sequences  : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   comments on tables     : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   comments on udts       : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   daos                   : false
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   deprecated code        : false
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   global references (any): true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   global references (catalogs): true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   global references (keys): true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   global references (links): true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   global references (queues): true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   global references (routines): true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   global references (schemas): true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   global references (sequences): true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   global references (tables): true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   global references (udts): true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   indexes                : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   instance fields        : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   interfaces             : false
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   interfaces (immutable) : false
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   javadoc                : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   keys                   : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   links                  : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   pojos                  : true (forced to true because of <immutablePojos/>)
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   pojos (immutable)      : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   queues                 : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   records                : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   routines               : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   sequences              : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   table-valued functions : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   tables                 : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   udts                   : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:   relations              : true
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO: ----------------------------------------------------------
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:                          
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO: Generation remarks       
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO: ----------------------------------------------------------
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO:                          
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO: ----------------------------------------------------------
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO: Generating catalogs      : Total: 1
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO: ARRAYs fetched           : 0 (0 included, 0 excluded)
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO: Enums fetched            : 0 (0 included, 0 excluded)
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO: Packages fetched         : 0 (0 included, 0 excluded)
Jul 15, 2018 4:08:15 PM org.jooq.tools.JooqLogger info
INFO: Routines fetched         : 0 (0 included, 0 excluded)
Exception in thread "main" java.lang.NoSuchMethodError: org.jooq.SelectJoinStep.where(Lorg/jooq/Condition;)Lorg/jooq/SelectConditionStep;
        at org.jooq.meta.sqlite.SQLiteDatabase.getTables0(SQLiteDatabase.java:317)
        at org.jooq.meta.AbstractDatabase.getTables(AbstractDatabase.java:1274)
        at org.jooq.codegen.JavaGenerator.generateSchemaIfEmpty(JavaGenerator.java:419)
        at org.jooq.codegen.JavaGenerator.generateCatalogIfEmpty(JavaGenerator.java:405)
        at org.jooq.codegen.JavaGenerator.generate(JavaGenerator.java:379)
        at org.jooq.codegen.GenerationTool.run(GenerationTool.java:750)
        at org.jooq.codegen.GenerationTool.generate(GenerationTool.java:220)
        at org.jooq.codegen.GenerationTool.main(GenerationTool.java:192)

> Task :generateSDEJooqSchemaSource FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':generateSDEJooqSchemaSource'.
> Process 'command '/Library/Java/JavaVirtualMachines/jdk-10.0.1.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
1 actionable task: 1 executed

```

I'm uncertain whether to post it here or on JOOQ, however as I'm using this plugin I thought it best to post here first.

Any help would be appreciated.

**Comment by lukaseder** [comment]
It appears that you're using jOOQ codegen / meta version 3.11.3 to generate code, but have an older version of the jOOQ core library on the classpath

**Comment by pskaarup** [comment]
If I exclude everything related to spring it works, so the solution for me is to downgrade jooq/struder until it matches what is bundled with `spring-boot-starter-jooq` or create a sub configuration for generator.

I think for now we can close the ticket as this is a miss-configuration and not issue with jooq/struder

**Comment by pskaarup** [comment]
Incase anyone runs into the same issue. I fixed it by downgrading jooq to 3.10.7 and nu.struder.jooq to 2.0.11
